### PR TITLE
Adding tool developer guide

### PR DESCRIPTION
**Pull request by cmungall**

### CODE REVIEW

**Comment by cmungall** [comment]
@ddooley - if you have time to look this over that would be great!

**Comment by ddooley** [comment]
Ok I will check it out.  Completely packed day though today.

File: sphinx/developers/tool-developer-guide.rst [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+
+There are a few broad approaches here:
+
+- 1. Use the direct output of your language's YAML/JSON parser, and
+  implement logic ad-hoc
+- 2. Develop a full blown analog of the python SchemaView class for
+  your language
+- 3. Write python code to transform LinkML YAML into a native form
+  geared for your application
+
```
Comment (ddooley) [inline sphinx/developers/tool-developer-guide.rst:145]: I think there is a fourth option, which we're doing in DataHarmonizer: Use LinkML python SchemaView and other calls to create a compiled data structure which can be read as YAML or JSON, which is then read-in by your platform and language of choice (e.g. javascript in a browser).

**Comment by cmungall** [comment]
That's actually what I intended by option 3 -- "native form" == "compiled
data structure" - maybe I need to make that clearer!

**Pull request merged**